### PR TITLE
Change setting.binary depending on a job tag

### PR DESCRIPTION
**Pull request by danixeee**

### CODE REVIEW

File: packages/nexrender-core/src/helpers/autofind.js [hunk 1]
```diff
@@ -92,6 +93,7 @@ module.exports = settings => {
     const binary  = 'aerender' + (platform === 'win32' || platform === 'wsl' ? '.exe' : '' )
     const results = defaultPaths[platform]
         .map(folderPath => path.join(folderPath, binary))
+        .filter(binaryPath => binaryPath.indexOf(versionYear || '') >= 0)
```
Comment (ivansenic) [inline packages/nexrender-core/src/helpers/autofind.js:96]: can an empty char match the binary path in any case, guess not?

File: packages/nexrender-core/src/index.js [hunk 2]
```diff
@@ -102,6 +102,8 @@ const init = (settings) => {
 
     // attempt to patch the default
     // Scripts/commandLineRenderer.jsx
+
+    // TODO: Patch in all existing AE dirs
     patch(settings);
```
Comment (ivansenic) [inline packages/nexrender-core/src/index.js:107]: what this means?

File: packages/nexrender-core/src/tasks/setup.js [hunk 3]
```diff
@@ -61,5 +63,22 @@ P.S. to prevent nexrender from removing temp file data, you also can please prov
 
     settings.logger.log(`[${job.uid}] working directory is: ${job.workpath}`);
 
+    // update aebinary to match version from the job
+    try {
+      const aeVersionYear = job.tags
+                                .filter(tag => tag.startsWith("AE"))
+                                .map(tag => new RegExp(/AE(\d+)/).exec(tag)[1])[0];
```
Comment (ivansenic) [inline packages/nexrender-core/src/tasks/setup.js:70]: what happens here for the `AENA` tag, this regex will not match, would version year be `undefined`?

File: packages/nexrender-core/src/tasks/setup.js [hunk 4]
```diff
@@ -61,5 +63,22 @@ P.S. to prevent nexrender from removing temp file data, you also can please prov
...
+      const aeBinary = autofind(settings, aeVersionYear);
```
Comment (ivansenic) [inline packages/nexrender-core/src/tasks/setup.js:71]: `AE2022` would extract `2022`, would this be in the path? should it only get the last two digits `22`?

File: packages/nexrender-core/src/tasks/setup.js (hunk 3 above)
Comment (ivansenic) [inline packages/nexrender-core/src/tasks/setup.js:70]: how would this work if I have a tags like `[useast1, AE2022]`, should be more safe imo? aha OK, will skip due to that filter..

File: packages/nexrender-core/src/tasks/setup.js [hunk 5]
```diff
@@ -61,5 +63,22 @@ P.S. to prevent nexrender from removing temp file data, you also can please prov
...
+      if (fs.existsSync(aeBinary)) {
+        settings.binary = aeBinary;
+        settings.logger.log(`[${job.uid}] Detected AE version tag, setting binary to: ${aeBinary}`);
+      } else {
+        settings.logger.log(`[${job.uid}] AE ${aeVersionYear} is not installed, keeping default binary: ${settings.binary}`);
+      }
+    } catch {
+      // AENA tag falls here
+      settings.logger.log(`[${job.uid}] AE version tag missing, keeping default binary: ${settings.binary}`);
+    }
```
Comment (ivansenic) [inline packages/nexrender-core/src/tasks/setup.js:81]: should not fail for not matching tags, but rather have `aeVersionYear` undefined and thus skip the find..

**Review by ivansenic — COMMENTED** [review]
I like the idea, very very interesting and good approach..

File: packages/nexrender-core/src/index.js (hunk 2 above)
Comment (danixeee) [reply packages/nexrender-core/src/index.js:109]: When we start the worker with `nexrender-worker --binary /path/to/ae2020/aerender` it replaces some script inside `/path/to/ae2020/Scripts` with the one from `nexrender-core/src/assets`. That's why it needs to be run using sudo the first time.
I will probably leave this TODO, the only thing we need to do is to call it for each installed AE version, like:

- nexrender-worker --binary /path/to/ae2020/aerender
- nexrender-worker --binary /path/to/ae2021/aerender
- nexrender-worker --binary /path/to/ae2022/aerender

File: packages/nexrender-core/src/tasks/setup.js (hunk 3 above)
Comment (danixeee) [reply packages/nexrender-core/src/tasks/setup.js]: in that case the default aerender is used:

nexrender-worker --binary /path/to/ae2020/aerender

so ae2020 in that case ^

File: packages/nexrender-core/src/tasks/setup.js (hunk 4 above)
Comment (danixeee) [reply packages/nexrender-core/src/tasks/setup.js]: yes, these are the paths: https://github.com/plainly-videos/nexrender/blob/master/packages/nexrender-core/src/helpers/autofind.js#L5

File: packages/nexrender-core/src/tasks/setup.js (hunk 3 above)
Comment (danixeee) [reply packages/nexrender-core/src/tasks/setup.js]: yes, but actually tags should be a string and we are passing a list. Once we fix that, we need `job.tags.split(',')...`

File: packages/nexrender-core/src/tasks/setup.js (hunk 5 above)
Comment (danixeee) [reply packages/nexrender-core/src/tasks/setup.js]: did some small refactoring and it works like that now

**Comment by danixeee** [comment]
@ivansenic I am still not able to log in to the staging PC and test it, but it works fine on mine macos

**Pull request merged**